### DISCUSSION TITLE
Things that get thrown off-station should eventually be deleted.

### DISCUSSION THREAD

**Discussion by PJB3005** [space-wizards/space-station-14#3969] [Ideaguys]
See #565.

If you throw something into space it'll keep flying forever. It would be good if things thrown in this manner were *somehow* stopped.

For now the easiest way would just be flat deletion. In the future we could maybe make it so that things you throw off-map gravitate towards some form of debris field where they get stopped.

**Comment by SweptWasTaken** [comment]
I don't think flat-out deleting them would be a good idea as there's always an off-chance it would be important enough to want to be recovered like the nuke disk or a mission critical item. (Although I know that in SS14 we'll obviously do something to make stuff like this impossible since its the equivalent of deleting the bomb in CSGO.)

The main problem we had with CM was people shooting bullets into infinite space, but there's also the point to be made that those have gameplay value as well, like imagine you're traveling towards the station in the arrivals shuttle and a bunch of bullets suddenly slam into you because some traitor had the foresight to spray lead where the shuttle was coming from. (Bad Example)

I'm not really an engine guy but could we do something like decreasing the amount of calculations on stuff that flies out of range of the station? 

Or maybe stuff slowly decelerates and drifts back towards the station?

**Comment by PJB3005** [comment]
> I don't think flat-out deleting them would be a good idea as there's always an off-chance it would be important enough to want to be recovered like the nuke disk or a mission critical item. 

The whole point is that it's basically impossible to recover them without admin help when they're just thrown into space like that.

> like the nuke disk or a mission critical item. 

SS13 automatically respawns the disk if it gets deleted.

> The main problem we had with CM was people shooting bullets into infinite space

No, the main problem was items being thrown into space. Bullets despawned after a bit. Items did not.

**Comment by SweptWasTaken** [comment]
> The whole point is that it's basically impossible to recover them without admin help when they're just thrown into space like that.

So in SS14 should we fix this? Because we might as well just delete them if they're going to be impossible to retrieve without admin intervention. (This is not the ideal option)

> SS13 automatically respawns the disk if it gets deleted.

Does SS14 have this functionality? Should I make this issue a feature to implement this functionality? 

> No, the main problem was items being thrown into space. Bullets despawned after a bit. Items did not.

My bad

**Comment by PJB3005** [comment]
> (This is not the ideal option)

This is why I suggested an alternative at the bottom of the OP...

> Does SS14 have this functionality? Should I make this issue a feature to implement this functionality?

Well we don't exactly have a nuke disk...

**Comment by ProfanedBane** [comment]
I personally like the debris field idea, could even have it be fluffed as a bluespace anomaly things end up drifting to in nearby space. Could have stuff from previous rounds turn up there etc.

**Comment by SweptWasTaken** [comment]
#705 Is somewhat related

I'm wondering if we should do something like throwing stuff in space makes it travel slower like in real life, maybe just slow shit down that's thrown into space making recoverability a little easier.

**Comment by ghost** [comment]
torus topology so items wrap around and hit the other side of the station when

**Comment by SweptWasTaken** [comment]
Fire bullets into space from one side of the station then run over to the other side to eat your own lead

**Comment by PrPleGoo** [comment]
I also like the items looping around the station SS13-esque.
What does it mean to be "off-station"?

**Comment by ShadowCommander** [comment]
The easiest way to define off-station is not being on a grid or shuttle. Though for things that aren't moving it's not so much of a problem, so for this issue it would be things not on a grid that are moving.

**Comment by PJB3005** [comment]
> I also like the items looping around the station SS13-esque.

Well the thing is if I shoot something out of the station mass driver I expect it to *stay* off station, not hit the windows on the other side of the station after 5 minutes.

**Comment by PrPleGoo** [comment]
Usually in SS13 it has to not get stuck on several different z-levels before ending up on the other side of the station.

Are there any plans for recreating these "nearby" structures that randomly border each other?
If not, then I agree that just looping around is weird and dumb.
If yes, then passing into these different zones until it gets stuck on something seems fine to me even if it could lead to it ending up on the other side of the station.
